app/models/viewmodels/decoder/ROECode.scala

```diff
@@ -16,24 +16,47 @@
 
 package models.viewmodels.decoder
 
+import play.api.Logger
+import play.api.libs.json.{Format, JsResult, JsString, JsSuccess, JsValue}
+
 /** ROE codes mapping based on Inventory Linking Exports codes.
   * Details can be found in Exports Notifications Behaviour sheet.
   *
   * @param code the code value
   * @param messageKey messages key with related description
   */
-sealed abstract class ROECode(override val code: String, override val messageKey: String) extends CodeWithMessageKey
+sealed abstract class ROECode(override val code: String, override val messageKey: String, val priority: Int) extends Ordered[ROECode] with CodeWithMessageKey {
+
+  override def compare(that: ROECode): Int = this.priority - that.priority
+}
 
 object ROECode {
 
+  private val logger = Logger(this.getClass)
+
   val codes: Set[ROECode] =
-    Set(DocumentaryControl, PhysicalExternalPartyControl, NonBlockingDocumentaryControl, NoControlRequired, RiskingNotPerformed, PrelodgePrefix)
+    Set(DocumentaryControl, PhysicalExternalPartyControl, NonBlockingDocumentaryControl, NoControlRequired, RiskingNotPerformed, PrelodgePrefix, UnknownRoe)
+
+  case object DocumentaryControl extends ROECode(code = "1", messageKey = "decoder.roe.DocumentaryControl", priority = 2)
+  case object PhysicalExternalPartyControl extends ROECode(code = "2", messageKey = "decoder.roe.PhysicalExternalPartyControl", priority = 1)
+  case object NonBlockingDocumentaryControl extends ROECode(code = "3", messageKey = "decoder.roe.NonBlockingDocumentaryControl", priority = 3)
+  case object NoControlRequired extends ROECode(code = "6", messageKey = "decoder.roe.NoControlRequired", priority = 6)
+  case object RiskingNotPerformed extends ROECode(code = "0", messageKey = "decoder.roe.RiskingNotPerformed", priority = 4)
+  case object PrelodgePrefix extends ROECode(code = "H", messageKey = "decoder.roe.PrelodgePrefix", priority = 5)
+  case object UnknownRoe extends ROECode(code = "", messageKey = "", priority = 100)
+
 
-  case object DocumentaryControl extends ROECode(code = "1", messageKey = "decoder.roe.DocumentaryControl")
-  case object PhysicalExternalPartyControl extends ROECode(code = "2", messageKey = "decoder.roe.PhysicalExternalPartyControl")
-  case object NonBlockingDocumentaryControl extends ROECode(code = "3", messageKey = "decoder.roe.NonBlockingDocumentaryControl")
-  case object NoControlRequired extends ROECode(code = "6", messageKey = "decoder.roe.NoControlRequired")
-  case object RiskingNotPerformed extends ROECode(code = "0", messageKey = "decoder.roe.RiskingNotPerformed")
-  case object PrelodgePrefix extends ROECode(code = "H", messageKey = "decoder.roe.PrelodgePrefix")
+  implicit object ROECodeFormat extends Format[ROECode] {
```

Comment:
Why do we need the json formatter?

Reply:
Because it's a part of the model that we're using for communication with the backend

Reply:
This is why I said before it's slightly weird to reuse actual model, because viewmodels with formatter and everything can be not a good idea

But still probably there is more cons than maintaining two models

Reply:
I see.